service/collector.go

```diff
@@ -154,6 +154,21 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	return nil
 }
 
+func (col *Collector) reloadConfiguration(ctx context.Context) error {
```

Comment:
I think we should make this no-op if the retrieved config is same as current config. Maybe we can add that later, but would be good to have it.

Reply:
I'm in favour of making this enhancement at a later time.